### PR TITLE
Reduce CI running time by reducing the use of the big dataset wikitext103

### PR DESCRIPTION
**Pull request by xuhdev**
Also use local http server for the invalid tarball test

### CODE REVIEW

**Comment by xuhdev** [comment]
Each run is shortened from 3:20 -> 2:30

File: tests/test_dataset.py [hunk 1]
```diff
@@ -155,15 +153,15 @@ def test_symlink_data_dir(self, tmp_symlink_dir, gmb_schema):
         dataset = Dataset(gmb_schema, data_dir=tmp_symlink_dir, mode=Dataset.InitializationMode.LAZY)
         assert dataset._data_dir == tmp_symlink_dir
 
-    def test_is_downloaded(self, tmp_path, wikitext103_schema):
+    def test_is_downloaded(self, tmp_path, gmb_schema):
```
Comment (edwardleardi) [inline tests/test_dataset.py:156]: didn't realize wikitext is 180mb vs 10mb for gmb, good idea

File: tests/conftest.py [hunk 2]
```diff
@@ -113,11 +113,20 @@ def schema_file_http_url(local_http_server) -> str:
...
         yield Dataset(wikitext103_schema, data_dir=tmp_data_dir, mode=Dataset.InitializationMode.DOWNLOAD_ONLY)
 
 
+@pytest.fixture
+def downloaded_gmb_dataset(gmb_schema) -> Dataset:
+    with TemporaryDirectory() as tmp_data_dir:
+        yield Dataset(gmb_schema, data_dir=tmp_data_dir, mode=Dataset.InitializationMode.DOWNLOAD_ONLY)
+
+
+# Temporary hacks -------------------------------------
```
Comment (edwardleardi) [inline tests/conftest.py:127]: nice 😆

**Review by edwardleardi — APPROVED** [review]
LGTM

**Pull request merged**